pkg/handler/repositories.go

```diff
@@ -201,13 +222,25 @@ func (rh *RepositoryHandler) bulkCreateRepositories(c echo.Context) error {
 	var msg *message.IntrospectRequestMessage
 	var err error
 	for _, repo := range responses {
-		if msg, err = adapter.NewIntrospect().FromRepositoryResponse(&repo); err != nil {
-			log.Error().Msgf("bulkCreateRepositories could not map to IntrospectRequest message: %s", err.Error())
-			continue
-		}
-		if err = rh.IntrospectRequestProducer.Produce(c, msg); err != nil {
-			log.Error().Msgf("bulkCreateRepositories returned an error: %s", err.Error())
-			continue
+		if config.Get().NewTaskingSystem {
+			task := queue.Task{
+				Typename: tasks.Introspect,
+				Payload:  tasks.IntrospectPayload{Url: repo.URL},
+				OrgId:    orgID,
+			}
+			_, err := rh.TaskClient.Enqueue(task)
+			if err != nil {
+				log.Error().Msgf("error enqueuing task: %s", err.Error())
+			}
+		} else {
+			if msg, err = adapter.NewIntrospect().FromRepositoryResponse(&repo); err != nil {
+				log.Error().Msgf("bulkCreateRepositories could not map to IntrospectRequest message: %s", err.Error())
+				continue
+			}
+			if err = rh.IntrospectRequestProducer.Produce(c, msg); err != nil {
+				log.Error().Msgf("bulkCreateRepositories returned an error: %s", err.Error())
+				continue
+			}
```

Comment:
I think you could extract this big block into a helper function right? it looks quite a bit duplicated from the non-bulk call

Reply:
and update :)

Reply:
good idea!